R/Figures.R

```diff
@@ -1138,6 +1138,385 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
+  
+  # if left blank, ensure comparable data identifier is in the id_cols vector
+  if (is.null(id_cols) | length(unique(id_cols)) == 1) {
+    id_cols <- c("TADA.ComparableDataIdentifier","TADA.ComparableDataIdentifier")
+  }
+  id_cols <- c("TADA.ComparableDataIdentifier", vecsets::vsetdiff(id_cols, "TADA.ComparableDataIdentifier"))
+  
+  # if id_cols are specified by user and the comparable data identifier is not selected.
+  if (!"TADA.ComparableDataIdentifier" %in% id_cols) {
+    stop("Note: TADA.ComparableDataIdentifier not found in id_cols argument.")
+  }
+  
+  # Only allows for 2 column selections in the id_cols vector, with TADA.ComparableDataIdentifier being a required value
+  if (length(id_cols) > 2) {
+    stop("id_cols argument can only be a maximum of length two.")
+  }
+  
+  # Runs if more than 1 Tada.ComparableDataIdentifiers are found in a dataframe. Prompts for user selection of a  single characteristic in the dataframe
+  # var <- NULL
+  # id2 <- NULL
+  # if(length(unlist(unique(.data[, id_cols[1]]))) > 1){
+  #   var <- utils::menu(unlist(unique(.data[, id_cols[1]])), title = "Please specify Characteristic to plot: ")
+  #   assign("id2", unlist(unique(.data[, id_cols[1]]))[var])
+  # }
+  
+  # if groups are not specified, select the top 4 groups to plot.
+  if (length(id_cols) == 2 & is.null(groups)) {
+    assign("groups", names((utils::head(sort(table(.data[,id_cols[2]]), decreasing = TRUE), 4))))
+    print("Note: No 'groups' specified. Plotting up to top 4 'groups' based on the number of field counts")
+  }
+  
+  # check that groups are in id_cols
+  id <- unlist(unique(.data[, id_cols[2]]))
+  if (any(!groups %in% id)) {
+    stop("The 'groups' vector contains one or more inputs that are not found within your input dataset. Check spelling and try again.")
+  }
+  
+  depthcols <- names(.data)[grepl("DepthHeightMeasure", names(.data))]
+  depthcols <- depthcols[grepl("TADA.", depthcols)]
+  
+  plot.data <- as.data.frame(.data)
+  
+  # this subset must include all fields included in plot hover below
+  plot.data <- subset(plot.data, plot.data[, id_cols[2]] %in% groups)[, c(id_cols, reqcols, depthcols, "ActivityStartDateTime", "MonitoringLocationName", "TADA.ActivityMediaName", "ActivityMediaSubdivisionName", "ActivityRelativeDepthName", "TADA.CharacteristicName", "TADA.MethodSpeciationName", "TADA.ResultSampleFractionText")]
+  # if(!is.null(var)){
+  #   plot.data <- subset(plot.data, plot.data[, "TADA.ComparableDataIdentifier"] %in% id2)
+  # }
+  
+  plot.data <- dplyr::arrange(plot.data, ActivityStartDate)
+  
+  # Returns the param groups for plotting. Up to 4 params are defined.
+  param1 <- param2 <- parm3 <- param4 <- NULL
+  for(i in 1:length(unique(groups))) {
+    assign(paste0("param",as.character(i)), subset(plot.data, plot.data[, id_cols[2]] %in% groups[i]))
+  }
+  
+  all_scatterplots <- list()
+  
+  for (i in 1:length(unique(plot.data$TADA.ComparableDataIdentifier))) {
```

Comment:
Given that so much of this is dependent on TADA.ComparableDataIdentifier, it seems to make the most sense to revise function to only require user to input the non-TADA.ComparableDataIdentifier id_col and its groups.